src/Blog/Post/PostController.php

```diff
@@ -2,28 +2,171 @@
 
 namespace App\Blog\Post;
 
+use App\Blog\Entity\Post;
+use App\Entity\User;
 use App\ViewRenderer;
+use Cycle\ORM\ORMInterface;
+use Cycle\ORM\Transaction;
 use Psr\Http\Message\ResponseFactoryInterface;
 use Psr\Http\Message\ResponseInterface as Response;
 use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Log\LoggerInterface;
+use Yiisoft\Access\AccessCheckerInterface;
+use Yiisoft\Http\Method;
+use Yiisoft\Router\UrlGeneratorInterface;
+use Yiisoft\Yii\Web\User\User as UserComponent;
 
 final class PostController
 {
     private ViewRenderer $viewRenderer;
+    private ResponseFactoryInterface $responseFactory;
+    private LoggerInterface $logger;
+    private UserComponent $userComponent;
 
-    public function __construct(ViewRenderer $viewRenderer)
-    {
+    public function __construct(
+        ViewRenderer $viewRenderer,
+        ResponseFactoryInterface $responseFactory,
+        LoggerInterface $logger,
+        UserComponent $userComponent
+    ) {
         $this->viewRenderer = $viewRenderer->withControllerName('blog/post');
+        $this->responseFactory = $responseFactory;
+        $this->logger = $logger;
+        $this->userComponent = $userComponent;
     }
 
-    public function index(Request $request, PostRepository $postRepository, ResponseFactoryInterface $responseFactory): Response
+    public function index(Request $request, PostRepository $postRepository, AccessCheckerInterface $accessChecker): Response
     {
+        $userId = $this->userComponent->getId();
+        $canEdit = !is_null($userId) && $accessChecker->userHasPermission($userId, 'editPost');
+
         $slug = $request->getAttribute('slug', null);
         $item = $postRepository->fullPostPage($slug);
         if ($item === null) {
-            return $responseFactory->createResponse(404);
+            return $this->responseFactory->createResponse(404);
+        }
+
+        return $this->viewRenderer->render('index', ['item' => $item, 'canEdit' => $canEdit, 'slug' => $slug]);
+    }
+
+    public function add(
+        Request $request,
+        ORMInterface $orm,
+        UrlGeneratorInterface $urlGenerator
+    ): Response {
+        if ($this->userComponent->isGuest()) {
+            return $this->responseFactory->createResponse(403);
+        }
+
+        $body = $request->getParsedBody();
+        $parameters = [
+            'body' => $body,
+            'action' => ['blog/add']
+        ];
+
+        if ($request->getMethod() === Method::POST) {
+            $error = '';
+
+            try {
+                foreach (['header', 'content'] as $name) {
```

Comment:
It is called "title".

```suggestion
                foreach (['title', 'content'] as $name) {
```